AFQ/tasks/segmentation.py

```diff
@@ -47,9 +54,17 @@ def segment(dwi, data_imap, mapping_imap,
     reg_template = data_imap["reg_template"]
     streamlines = tractography_imap["streamlines"]
     img = nib.load(dwi)
-    tg = load_tractogram(
-        streamlines, img, Space.VOX,
-        bbox_valid_check=False)
+    if streamlines.endswith(".trk"):
+        tg = load_tractogram(
+            streamlines, img, Space.VOX,
+            bbox_valid_check=False)
+        is_trx = False
+    elif streamlines.endswith(".trx"):
+        is_trx = True
+        trx = load_trx(streamlines, img)
+        trx.streamlines._data = trx.streamlines._data.astype(np.float32)
```

Comment:
Is it necessary to cast to float32 here? If the trx is saved at float16 this would double memory usage of this without increasing precision.

Reply:
this was so it would play nice with certain dipy cipy functions which I guess assume float32. We will have to make these functions compatible with float16 in the long run, but I figured this was good for now.

Reply:
Makes sense!